lib/Test/MockFile.pm

```diff
@@ -1141,7 +1141,7 @@ sub _abs_path_to_file {
 
     return q[/] if $path eq q[/..];
 
-    return $path if $path =~ m{^/};
+    return $path if $path =~ m{^[/~]}xms;
```

Comment:
alternatively could we consider substitute the `^~` by the `$ENV{HOME}` then go through the additional logic?

Reply:
why HOME? isn't this a cwd issue, not HOME?

Reply:
abs_path needs to be an absolute path. I'm good with $ENV{HOME} or even some lookup for getpwent. I'm concerned that `~/0` will ever be in the mocking index.

Reply:
I don't understand the concern about `~/0`.

Reply:
Having talked it over, I understand now. :)

Reply:
Pushed solution.